### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by thisisjoshford**

### CODE REVIEW

File: src/components/App/App.jsx [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+  BrowserRouter as Router,
+} from 'react-router-dom';
+import Main from '../../containers/Main';
+import ResultsList from '../../containers/SearchResultControls/SearchResultControls';
+import ArtistDetail from '../../containers/ArtistDetailControls/ArtistDetailControls';
 
 export default function App() {
-  return <h1>Hello World</h1>;
+  return (
+    <>
```
Comment (mehtaphysical) [inline src/components/App/App.jsx:13]: you can remove the fragment here

File: src/components/Search/Search.test.jsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
```
Comment (mehtaphysical) [inline src/components/Search/Search.test.jsx:1]: :+1:

File: src/containers/ArtistDetailControls/ArtistDetailControls.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React, { useState, useEffect } from 'react';
+import ArtistDetail from '../../components/ArtistDetail/ArtistDetail';
+import { useParams } from 'react-router-dom';
+import { getArtistDetail } from '../../services/apiFetches';
+
+const ArtistViewControl = () => {
+
```
Comment (mehtaphysical) [inline src/containers/ArtistDetailControls/ArtistDetailControls.jsx:7]: delete this line

File: src/components/ArtistDetail/ArtistDetail.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import withList from '../withList';
+
+const ArtistCard = ({ name }) => (
+  <p>{name}</p>
+);
+
+const ArtistDetail = withList(ArtistCard, 'artistDetail');
```
Comment (mehtaphysical) [inline src/components/ArtistDetail/ArtistDetail.jsx:9]: may not need withList here

File: src/containers/ArtistDetailControls/ArtistDetailControls.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  const [artistDetail, setArtistDetail] = useState([]);
```
Comment (mehtaphysical) [inline src/containers/ArtistDetailControls/ArtistDetailControls.jsx:8]: initial artistDetail should be `{}` or `null`

File: src/containers/Main.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  //redirect on submit to get those sweet params to share with your grandma :)
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    history.push(`/${artistName}`);
+  };
+
+  return (
+    <>
+      <h1>Artist Search</h1>
+      <Search onSubmit={handleSubmit} onChange={handleChange} />
```
Comment (mehtaphysical) [inline src/containers/Main.jsx:21]: pass artist name

File: src/components/3_ArtistDetail/ArtistDetail.jsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import styles from './ArtistDetail.css';
+
+
+const ArtistDetail = ({ artistName, artistDetail }) => {
+  const artistReleases = artistDetail.map((release) => (
+    <Link 
+      key={release.id} 
+      to={`/${artistName}/${release.title}/${release.id}/`} 
+    >
+      <div key={release.id} className={styles.Artist}>
```
Comment (janellemellor) [inline src/components/3_ArtistDetail/ArtistDetail.jsx:13]: Maybe consider using a more semantic element in place of div here

File: src/components/withList.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+const withList = (ListItemComponent, propName = 'list') => {
```
Comment (janellemellor) [inline src/components/withList.js:3]: nice use of withList- overall really clean, well organized code all around. Great job!

**Pull request merged**